src/Shared/LayoutRenderers/IISInstanceNameLayoutRenderer.cs

```diff
@@ -34,26 +35,42 @@ public class IISInstanceNameLayoutRenderer : LayoutRenderer
 #if ASP_NET_CORE
         private IHostEnvironment _hostEnvironment;
 
-        private IHostEnvironment HostEnvironment => _hostEnvironment ?? (_hostEnvironment = ServiceLocator.ResolveService<IHostEnvironment>(ResolveService<IServiceProvider>(), LoggingConfiguration));
-#endif
+        /// <summary>
+        /// Provides access to the current IHostEnvironment
+        /// </summary>
+        /// <returns>IHostEnvironment or <c>null</c></returns>
+        internal IHostEnvironment HostEnvironment
+        {
+            get => _hostEnvironment ?? (_hostEnvironment = RetrieveHostEnvironment(ResolveService<IServiceProvider>(), LoggingConfiguration));
+            set => _hostEnvironment = value;
+        }
 
+        internal static IHostEnvironment RetrieveHostEnvironment(IServiceProvider serviceProvider, LoggingConfiguration loggingConfiguration)
+        {
+            return ServiceLocator.ResolveService<IHostEnvironment>(serviceProvider, loggingConfiguration);
+        }
+#else
+        /// <summary>
+        /// Provides access to the current IHostEnvironment
+        /// </summary>
+        /// <returns>IHostEnvironment or <c>null</c></returns>
+        internal IHostEnvironment HostEnvironment { get; set; } = Internal.HostEnvironment.Default;
```

Comment:
Possible to merge the HostEnvironment-properties together, so the only difference is the method used to resolve the `IHostEnvironment`-implementation?